core/offchain/impl/offchain_worker_impl.cpp

```diff
@@ -92,7 +94,7 @@ namespace kagome::offchain {
     } catch (const std::system_error &exception) {
       return outcome::failure(exception.code());
     } catch (...) {
-      return outcome::failure(boost::system::error_code{});
+      return Error::OTHER_EXCEPTION;
```

Comment:
It appears to me that it might be extremely frustrating if it shows up in the logs as the reason of the node crash. I think it's better to define a specific error for offchain workers which would explain it its description that this is an error caused by some exception during the worker's thread start up, which we ignore due to the following reasons: *put reasons here*.

Reply:
Hm... I thought, this is exceptional situation, and **assert** should be enough here